### ISSUE TITLE
p:calendar Client Side Validation with DATE Pattern is not working (FIX included)

### ISSUE THREAD

**Issue by gako** [primefaces/primefaces#2013]
I found a bug in the validation.js File around line 460

```
if(pattern) {
	var patternArr = pattern.split(" ");
	for(var i = 0; i < patternArr.length; i++) {
		if(patternArr[i].toLowerCase().indexOf('h') !== -1) {
			timePattern = patternArr[i];
		}
		else if(patternArr[i].toLowerCase().indexOf('t') !== -1 && timePattern) {
			timePattern = timePattern + " " + patternArr[i];
		}
		else {
!!!			datePattern = patternArr[i];
		}
	}
}
```

The original pattern is e.g. "d M yy" after this parsing the variable datePattern only holds  the value "yy" because during each run of the loop the value is overwritten.
I think what you meant to do is something like this: (i only changed the line with ~~~~
```
if(pattern) {
	var patternArr = pattern.split(" ");
	for(var i = 0; i < patternArr.length; i++) {
		if(patternArr[i].toLowerCase().indexOf('h') !== -1) {
			timePattern = patternArr[i];
		}
		else if(patternArr[i].toLowerCase().indexOf('t') !== -1 && timePattern) {
			timePattern = timePattern + " " + patternArr[i];
		}
		else {
~~~			datePattern = datePattern ? datePattern + " "+ patternArr[i] : patternArr[i];
		}
	}
}
```

## 1) Environment
- PrimeFaces version: 6.0.13
- Application server + version: Glassfish 4.1.1 / Payara 4.1.1.163 #badassfish (build 215)
- Affected browsers: Chrome, Firefox

## 2) Expected behavior

Client Side Validation is performed correctly

## 3) Actual behavior

Validation always returns incorrect Format even for correct dates

## 4) Steps to reproduce

Create a client side validated form with an p:calendar componet and set the pattern to e.g. "d MMM yyyy"

**Comment by gako** [comment]
UPDATE1: Additional note, this error only occurs if the pattern contains "spaces" between the date parts. If the pattern is e.g. "d-MMM-yyyy" it works because the split method is doing nothing.

UPDATE2: I think the actual problem may really be that I use spaces as date part seperator between d,M and y. Because if only the date and time part is seperated by a space it works fine. I think the date/time part splitting has to be made more intelligent, don't know how exactly...

**Comment by mertsincan** [comment]
Fixed for 6.1.Final and 6.0.19. #2280
